Kernel/System/DynamicField/Driver/Set.pm

```diff
@@ -81,28 +81,98 @@ sub new {
     return $Self;
 }
 
+# Helper Function for getting the Dynamic Fields from an Include, i.e.
+# $DynamicFields = $GetIncludedDynamicFields->($Param{DynamicFieldConfig}{Config}{Include});
+# This subroutine takes two arguments:
+# $Include: a list of hash references containing information about the items to include
+# $DynamicFieldObject: an object used to retrieve dynamic field information
+# and returns either the DynamicFields or undef in case of an error.
+my $GetIncludedDynamicFields = sub {
+
+    my ($Include, $DynamicFieldObject, $DynamicFieldName) = @_;
+    my @DynamicFields;
+
+    # This subroutine takes a DFEntry and the DynamicFieldObject as arguments
+    # It retrieves the dynamic field definition for the given DFEntry
+    # If the definition is not available, it retrieves it from the DynamicFieldObject
+    # Returns the dynamic field definition
+    my $GetDynamicField = sub {
+
+        my ($DFEntry, $DynamicFieldObject) = @_;
+        
+        my $DynamicField = $DFEntry->{Definition} // $DynamicFieldObject->DynamicFieldGet(
+            Name => $DFEntry->{DF},
+        );
+
+        return $DynamicField;
+    };
+
+    my $Error = undef;
+
+    ITEM:
+    for my $IncludeItem (@{ $Include }) {
+
+        if ($IncludeItem->{Grid}) {
+
+            for my $Row (@{ $IncludeItem->{Grid}{Rows} }) {
+
+                for my $DFEntry (@$Row) {
+
+                    my $DynamicField = $GetDynamicField->($DFEntry, $DynamicFieldObject);
+                    if ($DynamicField) {
+                        push @DynamicFields, $DynamicField;
+                    } else {
+                        $Error = 1;
+                        last ITEM;
+                    }
+                }
+            }
+        } elsif ($IncludeItem->{DF}) {
+
+            my $DynamicField = $GetDynamicField->($IncludeItem, $DynamicFieldObject);
+            if ($DynamicField) {
+                push @DynamicFields, $DynamicField;
+            } else {
+                $Error = 1;
+                last ITEM;
+            }
+        } else {
+            $Error = 1;
+            last ITEM;
+        }
+    }
+
+    if ($Error) {
+        $Kernel::OM->Get('Kernel::System::Log')->Log(
+            Priority => 'error',
+            Message  => "$DynamicFieldName configured erroneously. 'DF: DynamicFieldName' missing or wrong.",
```

Comment:
You could set `$Error` to either "No DynamicField '$IncludeItem->{DF}'.", or "Need either Grid or DF in each Row.", or similar, based on the context, and then set the Message to `"$DynamicFieldName configured erroneously: $Error"` - in most cases probably not too important, as the "DynamicFieldGet()" would throw an error with the name of the field, but for missing Grid or DF probably more clear.